### PR TITLE
fix: Detect when the design of a view changed and update reference snapshot images

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

So whenever you made changes to a view. You needed manually to go to those tests add the isRecording = true and then run the tests. So the new tests could be re-generated. With this change every time the design of a view changes we will update reference snapshot images automatically. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift [hunk 1]
```diff
@@ -16,13 +16,16 @@
...
+import SnapshotTesting
 import WireCommonComponents
+import XCTest
 
 class BaseSnapshotTestCase: XCTestCase {
 
     override func setUp() {
         super.setUp()
+        // Enable when the design of the view has changed in order to update the reference snapshots
+        isRecording = strcmp(getenv("RECORDING_SNAPSHOTS"), "YES") == 0
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift:28]: > With this change every time the design of a view changes we will update reference snapshot images automatically.

How does this update the snapshots automatically? You still need to set the `RECORDING_SNAPSHOTS` environment var when you want to record snapshots.

Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift:28]: If you already have screenshots, you get the errors that the screenshots are not matching. When you delete those screenshots and run the tests it automatically record new ones.

**Review by netbe — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift:28]: Will new and old screenshots be shown in the PR?

Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift:28]: Yes, that won't change.

Comment (johnxnguyen) [reply wire-ios/Wire-iOS Tests/BaseSnapshotTestCase.swift:28]: Are you explicitly setting this environment var when you want to record new tests?

**Comment by agisilaos** [comment]
I'm going to close this since it's not fixing the issue we had with snapshots

**Pull request closed without merging**